lib/feature/recordings/presentation/pages/recordings_page.dart

```diff
@@ -24,13 +31,43 @@ class RecordingsPage extends StatelessWidget {
                 itemBuilder: (_, index) {
                   final item = state.models[index];
                   return ListTile(
-                    leading: Text(index.toString()),
-                    title: Text(item.title),
-                    subtitle: Text("Answered on ${item.model.questions.length} questions"),
-                    trailing: Icon(Icons.play_circle_fill),
-                    onTap: () {
-                      Navigator.push(context, MaterialPageRoute(builder: (_) => RecodingDetailsPage(item)));
-                    },
+                      leading: Text(index.toString()),
+                      title: Text(item.title),
+                      subtitle: Text("Answered on ${item.model.questions.length} questions"),
+                      trailing: Scaffold(
+                        persistentFooterButtons: <Widget>[
+                          PopupMenuButton<String>(
+                            initialValue: _selectedValue,
+                            onSelected: (String s) {
+                              setState(() {
+                                _selectedValue = s;
+                              });
+                            },
+                            itemBuilder: (BuildContext context) {
+                              return _usStates.map((String s) {
+                                return PopupMenuItem(
+                                  child: Text(s),
+                                  value: s,
+                                );
+                              }).toList();
+                            },
+                          )
+                        ],
+                        //    onTap: ()
+
+                      if (_selectedValue == "Play"){
+                        Navigator.push(context, MaterialPageRoute(builder: (_) => RecodingDetailsPage(item)));
+                      }else if (_selectedValue == "Delete"){
+                        builder: (BuildContext context, AsyncSnapshot<Tag> snapshot) {
+                      if (mapController.ready &&
+                        snapshot.hasData &&
+                        snapshot.data.mobile.nid != 0) {
+                        tag = snapshot.data;
+                        widget.tagStream.sink.add(Tag());
+                  }
+                  }
+                  },
+                  ),
```

Comment:
whole this new code is totally wrong 😢 
example solution:
```dart
PopupMenuButton(
                            itemBuilder: (context) => [
                                  PopupMenuItem(
                                    value: 1,
                                    child: Text(
                                      'Play',
                                    ),
                                  ),
                                  PopupMenuItem(
                                    value: 2,
                                    child: Text('Delete'),
                                  ),
                                ],
                            onSelected: (index) {
                              switch (index) {
                                case 1:
                                  // Play action
                                  break;
                                case 2:
                                  // delete acition
                                  break;
                              }
                            },
                            child: Icon(
                              Icons.more_vert,
                              size: 28,
                              color: Colors.white,
                            ))```

Reply:
I understood it.